Remove outdated/unnecessary files

rahulporuri opened the pull request:
There are a number of outdated/unnecessary files in the git source which can be removed. This PR removes some (not all) of them.

- kiva/agg/implementation_notes.txt has been moved to docs/kiva/agg
- enable/enable.txt has been removed
- enable/pyglet_backend/notes.txt has been removed
- enable/savage/compliance/w3c_svg_11 and enable/savage/compliance/w3c_svg_11/notes.txt have been removed. The enable/savage/complaince/README has been updated to provide the relevant information 
- enable/savage/readme.txt has been removed
- kiva/fonttools/tests/data/source.txt has been moved to the top level and renamed. At the same time, kiva/fonttools/tests/data/fonttools_LICENSE.txt has been moved to the top level. These files are now included in the source distribution via changes in the MANIFEST file.
- kiva/agg/src/swig_questions.txt has been removed
- kiva/agg/src/todo.txt has been removed.

jwiggins reviewed (approved):
What about any of these files?
```
enable/pyglet_backend/notes.txt
enable/savage/compliance/w3c_svg_11/notes.txt
enable/savage/readme.txt
docs/kiva/agg/interface.txt
kiva/agg/freetype2/docs/raster.txt
kiva/agg/freetype2/docs/formats.txt
kiva/agg/implementation_notes.txt
kiva/agg/src/todo.txt
kiva/agg/src/swig_questions.txt
kiva/fonttools/tests/data/source.txt
```

rahulporuri commented:
> docs/kiva/agg/interface.txt

It looked like useful information so i kept this around

>  kiva/agg/freetype2/docs/raster.txt
  kiva/agg/freetype2/docs/formats.txt

It looks like they're coming from freetype2 source and are not something we added - so i kept them around.

I didn't remove them 

> kiva/agg/implementation_notes.txt

I moved this into docs/kiva/agg as it looked like useful documentation.
